octopoes/octopoes/core/service.py

```diff
@@ -128,11 +128,20 @@ def _run_inference(self, origin: Origin, valid_time: datetime):
         parameters_references = self.origin_parameter_repository.list_by_origin(origin.id, valid_time)
         parameters = self.ooi_repository.get_bulk({x.reference for x in parameters_references}, valid_time)
 
+        resulting_oois = []
+
         try:
-            resulting_oois = BitRunner(bit_definition).run(source, list(parameters.values()))
-        except Exception as e:
-            logger.exception("Error running inference", exc_info=e)
-            return
+            level = self.scan_profile_repository.get(origin.source, valid_time).level
+        except ObjectNotFoundException:
+            level = 0
+
+        if level >= bit_definition.min_scan_level:
+            try:
+                resulting_oois = BitRunner(bit_definition).run(source, list(parameters.values()))
+            except Exception as e:
+                logger.exception("Error running inference", exc_info=e)
+                return
+
```

Comment:
@ammar92 

The logic here: when a level is not found (could be created with a slight delay, asynchronously), it should default to 0. Then proceed with the normal flow of checking the scan level and running the Bit